### PR TITLE
Arrow Binary Integration

### PR DESCRIPTION
**Pull request by pdet**
This PR implements the integration of arrow binary types to DuckDB's Blobs.

I've also enabled the parquet files with binary types for the tests.

### CODE REVIEW

File: test/arrow/parquet_test.cpp [hunk 1]
```diff
@@ -156,6 +167,7 @@ TEST_CASE("Test Parquet Files", "[arrow]") {
 
 	auto parquet_files = fs.Glob("test/sql/copy/parquet/data/*.parquet");
 	for (auto &parquet_path : parquet_files) {
+		std::cout << parquet_path << std::endl;
```
Comment (Mytherin) [inline test/arrow/parquet_test.cpp:170]: Debug code?

File: test/arrow/parquet_test.cpp [hunk 2]
```diff
@@ -166,17 +178,13 @@ TEST_CASE("Test Parquet Files", "[arrow]") {
...
 	}
-	//! FIXME: GLOB Files (More binaries)
-	//	parquet_files = fs.Glob("test/sql/copy/parquet/data/glob/*.parquet");
-	//	for (auto &parquet_path : parquet_files) {
-	//	    std::cout << parquet_path <<  std::endl;
-	//        REQUIRE(RoundTrip(parquet_path,skip,conn));
-	//	}
+	parquet_files = fs.Glob("test/sql/copy/parquet/data/glob/*.parquet");
+	for (auto &parquet_path : parquet_files) {
+		std::cout << parquet_path << std::endl;
```
Comment (Mytherin) [inline test/arrow/parquet_test.cpp:187]: Debug code?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good. Minor comment:

**Comment by pdet** [comment]
I've cleaned up the debug code, and added size checks for binary and strings ;-)

File: src/function/table/arrow.cpp [hunk 3]
```diff
@@ -304,6 +317,63 @@ void ArrowToDuckDBList(Vector &vector, ArrowArray &array, ArrowScanState &scan_s
...
+			if (FlatVector::IsNull(vector, row_idx)) {
+				continue;
+			}
+			auto bptr = cdata + offsets[row_idx];
+			auto blob_len = offsets[row_idx + 1] - offsets[row_idx];
+			FlatVector::GetData<string_t>(vector)[row_idx] = StringVector::AddStringOrBlob(vector, bptr, blob_len);
+		}
+	} else {
+		//! Check if last offset is higher than max uint32
+		if (((uint64_t *)array.buffers[1])[array.length] > std::numeric_limits<uint32_t>::max()) {
```
Comment (Mytherin) [inline src/function/table/arrow.cpp:358]: Use NumericLimits<uint32_t>::Maximum() to avoid upsetting Windows.

Comment (pdet) [reply src/function/table/arrow.cpp]: Yeah I keep forgetting that

**Pull request merged**